src/services/dashboards/widgets/aws-cloud-front-cost/AWSCloudFrontCost.vue

```diff
@@ -39,122 +45,217 @@ import {
-    tableFields: computed(() => [
-        GROUP_BY_ITEM_MAP[GROUP_BY.PROJECT],
-        { name: 'data-transfer.out', label: 'Transfer-Out' },
-        { name: 'requests.http', label: 'Requests (HTTP)' },
-        { name: 'requests.https', label: 'Requests (HTTPS)' },
-    ]),
+    ...toRefs(useWidgetState<FullData>(props)),
+    fieldsKey: computed<string>(() => SELECTOR_TYPE_FIELDS_KEY_MAP[state.selectedSelectorType]),
+    legends: [] as Legend[],
+    chartData: computed(() => {
+        const valueKey = `${state.fieldsKey}_sum`;
+        const _chartData = getRefinedXYChartData(state.data?.results, state.groupBy, 'usage_type', valueKey, true);
+        return reverse(_chartData);
+    }),
+    tableFields: computed<Field[]>(() => {
+        if (!state.groupBy) return [];
+        const textOptions: Field['textOptions'] = {
+            type: state.fieldsKey === 'usd_cost' ? 'cost' : 'size',
+            sourceUnit: USAGE_SOURCE_UNIT,
+        };
+        return [
+            GROUP_BY_ITEM_MAP[state.groupBy],
+            { name: `${state.fieldsKey}_sum.0.value`, label: 'Transfer-out', textOptions },
+            { name: `${state.fieldsKey}_sum.1.value`, label: 'Requests (HTTP)' },
+            { name: `${state.fieldsKey}_sum.2.value`, label: 'Requests (HTTPS)' },
+        ];
+    }),
+    dateRange: computed<DateRange>(() => {
+        const end = state.settings?.date_range?.end ?? dayjs.utc().format('YYYY-MM');
+        const range = props.size === WIDGET_SIZE.full ? 11 : 3;
+        const start = dayjs.utc(end).subtract(range, 'month').format('YYYY-MM');
+        return { start, end };
+    }),
+    thisPage: 1,
 });
 
 const widgetFrameProps:ComputedRef = useWidgetFrameProps(props, state);
 
 /* Api */
-const fetchData = async () => new Promise((resolve) => {
-    setTimeout(() => {
-        resolve(SAMPLE_RAW_DATA.results);
-    }, 1000);
-});
+const fetchData = async (): Promise<FullData> => {
+    if (!state.groupBy) return { results: [], more: false };
+    try {
+        const apiQueryHelper = new ApiQueryHelper();
+        apiQueryHelper.setFilters([
+            { k: 'usage_type', v: ['data-transfer.out', 'requests.http', 'requests.https'], o: '' },
+            { k: 'product', v: 'AmazonCloudFront', o: '=' },
+            { k: 'usage_type', v: null, o: '!=' },
+        ]);
+        const query = {
+            granularity: state.granularity,
+            group_by: [state.groupBy, 'usage_type'],
+            start: state.dateRange.start,
+            end: state.dateRange.end,
+            fields: {
+                usd_cost_sum: {
+                    key: SELECTOR_TYPE_FIELDS_KEY_MAP.cost,
+                    operator: 'sum',
+                },
+                usage_quantity_sum: {
+                    key: SELECTOR_TYPE_FIELDS_KEY_MAP.usage,
+                    operator: 'sum',
+                },
+            },
+            sort: [{ key: '_total_usd_cost_sum', desc: true }],
```

Comment:
question: Isn't it replaced by apiQueryHelper.data? If it's not, just ignore this comment 😅

Reply:
apiQueryHelper has only `filters` :)